core/frontend/src/extension/providers/ServiceExtensionProvider.ts

```diff
@@ -0,0 +1,95 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { rcompare } from "semver";
+
+import { IModelApp } from "../../IModelApp";
+import { loadScript } from "./ExtensionLoadScript";
+import { ExtensionClient } from "./ExtensionServiceClient";
+
+import type {
+  ExtensionManifest, ExtensionProps, ExtensionProvider, ServiceExtensionProviderProps,
+} from "../Extension";
+
+/**
+ * Implements a "service" extension via LocalExtensionProps.
```

Comment:
Let's avoid the term "service" extension here as that's fairly confusing plus it doesn't look to use the `LocalExtensionProps`.

Reply:
@johnnyd710  and I both agree. We spoke about this, more in the context of the type and interface names, and decided to get others opinions in this PR. I could not come up with a meaningful way to express this type of extension. Everything that we came up with was not unique to this case.

Reply:
@calebmshafer what do we want to call Extensions that are hosted on the Extension Service?

Reply:
I'd just refer to it as, 

"Implements an Extension from the Extension Service via the ServiceExtensionProviderProps."

At least for now and we can figure out better naming in the next release.